### PR TITLE
Adding support for measure names in the dashboard

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Update comparisons to use the actual measure name instead of assuming `measure_` in the name.
- [x] Update metrics definition editor to recognise measure name
- [x] Update auto generation of dashboard to generate a sanitised sticky measure name
- [x] Add check for duplicate names in runtime reconciliation

### CODE REVIEW

**Review by begelundmuller — APPROVED** [review]

File: web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts [hunk 1]
```diff
@@ -89,6 +89,25 @@ export function getFilterForComparisonTable(
...
+        .replace("_delta_perc", ".2")
+    );
+    const b_num = parseFloat(
+      measure2
+        .replace(measureName, "")
+        .replace("_delta", ".1")
+        .replace("_delta_perc", ".2")
+    );
+
+    return a_num - b_num;
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts:107]: I don't think this works as expected now. Two issues - 

This was based on the fact that measure names will always have a number beside them and the sorting worked as such. But now that we don't have that flexibility, we need to add `someMeasureName_delta` and `someMeasureName_delte_perc` columns adjacent to `someMeasureName`. If the order returned by the `meta` of `topListQuery` can be trusted then we can use that as is and skip the sorting stage.

2. Also this doesn't sort the measures properly even if they are named `measure_0`, `measure_1` etc. There's a test case for `customSortMeasures` in `dimension-table-utils.spec.ts`. You can replace the old method with the new one.

Comment (AdityaHegde) [reply web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts]: I am getting rid of the sorting. Just `splice`ing the comparison columns just after the selected sort measure should work.

File: web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte [hunk 2]
```diff
@@ -271,24 +269,30 @@
...
+          `${sortByColumn}_delta_perc`
+        );
+      }
     }
-    columnNames = columnNames.sort(customMeasureSortGetter(sortByColumn));
 
     // Make dimension the first column
     columnNames.unshift(dimension?.name);
     measureNames = allMeasures.map((m) => m.name);
+    console.log(columnNames);
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte:295]: Let's get rid of this.

File: web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts [hunk 3]
```diff
@@ -89,25 +89,6 @@ export function getFilterForComparisonTable(
...
-        .replace(measureName, "")
-        .replace("_delta", ".1")
-        .replace("_delta_perc", ".2")
-    );
-
-    return a_num - b_num;
-  };
-}
-
 export function customSortMeasures(measure1: string, measure2: string) {
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts]: We can also remove this stale method now along with it's test code

**Review by djbarnwal — APPROVED** [review]
Looks good

There are stale comments and methods which should be removed. Approving so post removal can be merged.

**Pull request merged**